MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+    
+#     def find_index(self,entry):
+#         for i, item in self.list:
+#             if item==entry:
+#                 return i
+            
+#     def find_val(self,index):
+#         return self.list[index]
+
+#####################
+
+    def find_index(self,id):
+        return self.id_index[id]
+    
+    def find_val(self,id):
+        index = self.find_index(id)
+        return self.list[index]
+
+
+###################
+    def GetAll(self):
+        return self.list
+    
+    def delete(self, entry):
+        
+        self.list.remove(entry)
+    
+    def update(self,id, new_entry):
+        
+        index = self.id_index[id]
+        self.list[index] = new_entry
+        
+    
+
+class User(BaseModel):
+    
+    def __init__(self):
+        self.email_userid = {}
+        self.userid = 0
+        super().__init__()
+        
+    
+    def register(self, client_name, email, password):
+        entry = {'client_name' : client_name,
+                'email': email,
+                'password':password,
+                'userid': self.userid }
```

Comment:
`userid` field should be renamed to `id`. Always name the main primary key as `id` unless you have a really good reason not to.

Reply:
Also note that assigning the `id` field is common between models. See also line 122 for the similar code in the Contact model. Something tells me that it should not be a responsibility of child model classes.